Otus-DevOps-2018-02/ifqthenp_microservices
# HW 20. Monitoring of application and infrastructure (monitoring-2)

# Completed Homework №20


## The following tasks have been completed

- :large_blue_diamond: Installed Google's cAdvisor resource analyser on Docker container to collect metrics about services
- :large_blue_diamond: Installed and configured Grafana analytics platform for collected metrics
- :large_blue_diamond: Created and imported various dashboards in Grafana including histogram and percentile metrics from Prometheus
- :large_blue_diamond: Set up alerting rules for Alertmanager from Prometheus and configured Slack notifications
- :large_orange_diamond: Added rules for new images to the `Makefile`
- :large_orange_diamond: Set up experimental Docker daemon metrics collection in Prometheus format
- :large_orange_diamond: Set up InfluxDB Telegraf for Docker daemon metrics collection
- :large_orange_diamond: Configured Alertmanager to send email notifications

## How to get project running

- Create GCE instance using `docker-machine` command
```shell
docker-machine create --driver google \
--google-machine-image https://www.googleapis.com/compute/v1/projects/ubuntu-os-cloud/global/images/family/ubuntu-1604-lts \
--google-machine-type n1-standard-1 \
--google-disk-size 20 \
--google-zone europe-west2-a \
docker-host
```

- Build images using `Makefile`
- Run `docker-compose up -d` and `docker-compose -f docker-compose-monitoring.yml up -d` from `docker` folder

## How to check that project actually works

- Ensure that the build passed TravisCI tests
  - [#andrei_bogomja](https://devops-team-otus.slack.com/messages/C9M6L1XB3) Slack channel
  - actual GitHub pull request page
- Get GCE instance public IP with `docker-machine ip docker-host`
- Type URL in browser address bar `http://<PUBLIC_IP>:9090/targets` and make sure that all services have `UP` status

## PR checklist
